rest_framework_json_api/filters/filter.py

```diff
@@ -0,0 +1,135 @@
+            """
+            do nothing
+            """
+            return queryset
+
+else:
+    class JSONAPIDjangoFilter(DjangoFilterBackend):
+        """
+        A Django-style ORM filter implementation, using `django-filter`.
+
+        This is not part of the jsonapi standard per-se, other than the requirement
+        to use the `filter` keyword: This is an optional implementation of style of
+        filtering in which each filter is an ORM expression as implemented by
+        DjangoFilterBackend and seems to be in alignment with an interpretation of
+        http://jsonapi.org/recommendations/#filtering, including relationship
+        chaining. It also returns a 400 error for invalid filters.
+
+        Filters can be:
+        - A resource field equality test:
+            `?filter[qty]=123`
+        - Apply other [field lookup](https://docs.djangoproject.com/en/stable/ref/models/querysets/#field-lookups)  # noqa: E501
+          operators:
+            `?filter[name.icontains]=bar` or `?filter[name.isnull]=true...`
+        - Membership in a list of values:
+            `?filter[name.in]=abc,123,zzz (name in ['abc','123','zzz'])`
+        - Filters can be combined for intersection (AND):
+            `?filter[qty]=123&filter[name.in]=abc,123,zzz&filter[...]`
+        - A related resource path can be used:
+        `?filter[inventory.item.partNum]=123456 (where `inventory.item` is the relationship path)`
+
+        If you are also using rest_framework.filters.SearchFilter you'll want to customize
+        the name of the query parameter for searching to make sure it doesn't conflict
+        with a field name defined in the filterset.
+        The recommended value is: `search_param="filter[search]"` but just make sure it's
+        `filter[<something>]` to comply with the jsonapi spec requirement to use the filter
+        keyword. The default is "search" unless overriden but it's used here just to make sure
+        we don't complain about it being an invalid filter.
+        """
+        # TODO: find a better way to deal with search_param.
+        search_param = api_settings.SEARCH_PARAM
+
+        # Make this regex check for 'filter' as well as 'filter[...]'
+        # Leave other incorrect usages of 'filter' to JSONAPIQueryValidationFilter.
+        # See http://jsonapi.org/format/#document-member-names for allowed characters
+        # and http://jsonapi.org/format/#document-member-names-reserved-characters for reserved
+        # characters (for use in paths, lists or as delimiters).
+        # regex `\w` matches [a-zA-Z0-9_].
+        # TODO: U+0080 and above allowed but not recommended. Leave them out for now. Fix later?
+        #       Also, ' ' (space) is allowed within a member name but not recommended.
+        filter_regex = re.compile(r'^filter(?P<ldelim>\[?)(?P<assoc>[\w\.\-]*)(?P<rdelim>\]?$)')
+
+        def validate_filter(self, keys, filterset_class):
+            for k in keys:
```

Comment:
for validation whether a filter is used shouldn't we not also use the form?

See https://github.com/carltongibson/django-filter/blob/2.0.0/django_filters/filterset.py#L223

Reply:
Somehow I thought he dropped `filter_queryset` between 1.1 and 2.0 so I was overriding the 1.1 version of it. Now I see it's there in 2.0 as well, but not the one you reference but rather this one: https://github.com/carltongibson/django-filter/blob/2.0.0/django_filters/rest_framework/backends.py#L89-L96

I didn't need to override it in 2.0 because it just calls `get_filterset` which I did override. But in 1.1 there's no call to get_filterset: https://github.com/carltongibson/django-filter/blob/1.1.0/django_filters/rest_framework/backends.py#L50-L56

So I'll have to fix this to make sure it doesn't break 2.0! Probably just check the [VERSION]( https://github.com/carltongibson/django-filter/blob/1.1.0/django_filters/__init__.py#L34) and `return super(...)` if > 1.1.